src/ClientData/TimerDataTest.cpp

```diff
@@ -66,4 +68,92 @@ TEST(TimerData, AddTimers) {
   EXPECT_EQ(timer_data.GetMinTime(), 2);
 }
 
+TEST(TimerData, FindTimers) {
+  TimerData timer_data;
+
+  {
+    TimerInfo timer_info;
+    timer_info.set_start(2);
+    timer_info.set_end(5);
+    timer_data.AddTimer(0, timer_info);
+
+    timer_info.set_start(8);
+    timer_info.set_end(11);
+    timer_data.AddTimer(0, timer_info);
+
+    timer_info.set_start(10);
+    timer_info.set_end(12);
+    timer_data.AddTimer(0, timer_info);
+  }
+
+  {
+    const TimerInfo* timer_info = timer_data.GetFirstAfterTime(4, 0);
+    ASSERT_NE(timer_info, nullptr);
+    EXPECT_EQ(timer_info->start(), 8);
+    EXPECT_EQ(timer_info->end(), 11);
+  }
+
+  {
+    const TimerInfo* timer_info = timer_data.GetFirstAfterTime(2, 0);
+    ASSERT_NE(timer_info, nullptr);
+    EXPECT_EQ(timer_info->start(), 8);
+    EXPECT_EQ(timer_info->end(), 11);
+  }
+
+  {
+    const TimerInfo* timer_info = timer_data.GetFirstAfterTime(1, 0);
+    ASSERT_NE(timer_info, nullptr);
+    EXPECT_EQ(timer_info->start(), 2);
+    EXPECT_EQ(timer_info->end(), 5);
+  }
+
+  {
+    const TimerInfo* timer_info = timer_data.GetFirstAfterTime(9, 0);
+    ASSERT_NE(timer_info, nullptr);
+    EXPECT_EQ(timer_info->start(), 10);
+    EXPECT_EQ(timer_info->end(), 12);
+  }
+
+  {
+    const TimerInfo* timer_info = timer_data.GetFirstAfterTime(10, 0);
+    EXPECT_EQ(timer_info, nullptr);
+  }
+
+  {
+    const TimerInfo* timer_info = timer_data.GetFirstAfterTime(0, 1);
+    EXPECT_EQ(timer_info, nullptr);
+  }
+
+  {
+    const TimerInfo* timer_info = timer_data.GetFirstBeforeTime(6, 0);
+    ASSERT_NE(timer_info, nullptr);
+    EXPECT_EQ(timer_info->start(), 2);
+    EXPECT_EQ(timer_info->end(), 5);
+  }
+
+  {
+    const TimerInfo* timer_info = timer_data.GetFirstBeforeTime(4, 0);
```

Comment:
I would like to have an additional query for FirstBeforeTime that returns the last timer.

Reply:
done. and yes - there was a bug.